--- origsrc/tracker-miners-2.0.4/configure.ac	2018-02-06 17:36:54.000000000 -0600
+++ src/tracker-miners-2.0.4/configure.ac	2018-02-27 16:47:25.586819000 -0600
@@ -1506,7 +1506,7 @@ AC_ARG_ENABLE(libvorbis,
 
 if test "x$enable_libvorbis" != "xno" ; then
    PKG_CHECK_MODULES(LIBVORBIS,
-                     [vorbisfile >= $LIBVORBIS_REQUIRED],
+                     [vorbis vorbisfile >= $LIBVORBIS_REQUIRED],
                      [have_libvorbis=yes],
                      [have_libvorbis=no])
 
--- origsrc/tracker-miners-2.0.4/src/libtracker-extract/Makefile.am	2017-12-16 05:27:16.000000000 -0600
+++ src/tracker-miners-2.0.4/src/libtracker-extract/Makefile.am	2018-02-12 17:03:11.127374500 -0600
@@ -10,6 +10,7 @@ AM_CPPFLAGS =
 libtracker_extractdir = $(libdir)/tracker-miners-$(TRACKER_API_VERSION)
 libtracker_extract_LTLIBRARIES = libtracker-extract.la
 
+libtracker_extract_la_LDFLAGS = -release $(TRACKER_API_VERSION) -no-undefined
 libtracker_extract_la_LIBADD =   \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
 	$(BUILD_LIBS)                                  \
--- origsrc/tracker-miners-2.0.4/src/libtracker-miners-common/Makefile.am	2018-02-06 07:08:35.000000000 -0600
+++ src/tracker-miners-2.0.4/src/libtracker-miners-common/Makefile.am	2018-02-12 16:57:17.621191200 -0600
@@ -9,17 +9,8 @@ noinst_LTLIBRARIES = libtracker-miners-c
 
 libtracker_miners_common_la_SOURCES = \
 	$(BUILT_SOURCES) \
-	tracker-date-time.c \
-	tracker-dbus.c \
-	tracker-domain-ontology.c \
-	tracker-file-utils.c \
-	tracker-ioprio.c \
-	tracker-log.c \
-	tracker-sched.c \
 	tracker-seccomp.c \
-	tracker-type-utils.c \
-	tracker-utils.c \
-	tracker-locale.c
+	$(NULL)
 
 noinst_HEADERS = \
 	tracker-dbus.h \
@@ -36,7 +27,9 @@ noinst_HEADERS = \
 	tracker-utils.h \
 	tracker-locale.h
 
+libtracker_miners_common_la_LDFLAGS = -L$(libtracker_miners_commondir)
 libtracker_miners_common_la_LIBADD = \
+	-ltracker-common \
 	$(BUILD_LIBS) \
 	$(LIBTRACKER_COMMON_LIBS) \
 	-lm
--- origsrc/tracker-miners-2.0.4/src/tracker-extract/Makefile.am	2018-02-06 17:36:54.000000000 -0600
+++ src/tracker-miners-2.0.4/src/tracker-extract/Makefile.am	2018-02-12 16:57:17.621191200 -0600
@@ -3,7 +3,7 @@
 #
 rulesdir = $(TRACKER_EXTRACT_RULES_DIR)
 
-module_flags = $(DTAGS_LDFLAGS) -module -avoid-version -no-undefined
+module_flags = $(DTAGS_LDFLAGS) -module -avoid-version -no-undefined -Wl,libtracker-extract.dll.a
 
 AM_CPPFLAGS = \
 	$(BUILD_CFLAGS) \
@@ -198,6 +198,7 @@ rules_DATA += 10-ebooks.rule
 libextract_abw_la_SOURCES = tracker-extract-abw.c
 libextract_abw_la_CFLAGS = $(TRACKER_EXTRACT_MODULES_CFLAGS)
 libextract_abw_la_LDFLAGS = $(module_flags)
+libextract_abw_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_abw_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -208,6 +209,7 @@ libextract_abw_la_LIBADD = \
 libextract_mp3_la_SOURCES = tracker-extract-mp3.c
 libextract_mp3_la_CFLAGS = $(TRACKER_EXTRACT_MODULES_CFLAGS)
 libextract_mp3_la_LDFLAGS = $(module_flags)
+libextract_mp3_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_mp3_la_LIBADD =  \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -218,6 +220,7 @@ libextract_mp3_la_LIBADD =  \
 libextract_vorbis_la_SOURCES = tracker-extract-vorbis.c
 libextract_vorbis_la_CFLAGS = $(TRACKER_EXTRACT_MODULES_CFLAGS)
 libextract_vorbis_la_LDFLAGS = $(module_flags)
+libextract_vorbis_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_vorbis_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -229,6 +232,7 @@ libextract_vorbis_la_LIBADD = \
 libextract_flac_la_SOURCES = tracker-extract-flac.c
 libextract_flac_la_CFLAGS = $(TRACKER_EXTRACT_MODULES_CFLAGS)
 libextract_flac_la_LDFLAGS = $(module_flags)
+libextract_flac_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_flac_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -242,6 +246,7 @@ libextract_oasis_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS) \
 	$(LIBGSF_CFLAGS)
 libextract_oasis_la_LDFLAGS = $(module_flags)
+libextract_oasis_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_oasis_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -255,6 +260,7 @@ libextract_epub_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS) \
 	$(LIBGSF_CFLAGS)
 libextract_epub_la_LDFLAGS = $(module_flags)
+libextract_epub_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_epub_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -268,6 +274,7 @@ libextract_png_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS) \
 	$(LIBPNG_CFLAGS)
 libextract_png_la_LDFLAGS = $(module_flags)
+libextract_png_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_png_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -279,6 +286,7 @@ libextract_png_la_LIBADD = \
 libextract_ps_la_SOURCES = tracker-extract-ps.c
 libextract_ps_la_CFLAGS = $(TRACKER_EXTRACT_MODULES_CFLAGS)
 libextract_ps_la_LDFLAGS = $(module_flags)
+libextract_ps_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_ps_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -289,6 +297,7 @@ libextract_ps_la_LIBADD = \
 libextract_xmp_la_SOURCES = tracker-extract-xmp.c
 libextract_xmp_la_CFLAGS = $(TRACKER_EXTRACT_MODULES_CFLAGS)
 libextract_xmp_la_LDFLAGS = $(module_flags)
+libextract_xmp_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_xmp_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(BUILD_LIBS) \
@@ -300,6 +309,7 @@ libextract_html_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS) \
 	$(LIBXML2_CFLAGS)
 libextract_html_la_LDFLAGS = $(module_flags)
+libextract_html_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_html_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(BUILD_LIBS) \
@@ -312,6 +322,7 @@ libextract_msoffice_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS) \
 	$(LIBGSF_CFLAGS)
 libextract_msoffice_la_LDFLAGS = $(module_flags)
+libextract_msoffice_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_msoffice_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -325,6 +336,7 @@ libextract_msoffice_xml_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS) \
 	$(LIBGSF_CFLAGS)
 libextract_msoffice_xml_la_LDFLAGS = $(module_flags)
+libextract_msoffice_xml_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_msoffice_xml_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -337,6 +349,7 @@ libextract_xps_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS) \
 	$(LIBGXPS_CFLAGS)
 libextract_xps_la_LDFLAGS = $(module_flags)
+libextract_xps_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_xps_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -350,6 +363,7 @@ libextract_pdf_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS) \
 	$(POPPLER_CFLAGS)
 libextract_pdf_la_LDFLAGS = $(module_flags)
+libextract_pdf_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_pdf_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -362,6 +376,7 @@ libextract_dvi_la_SOURCES = tracker-extr
 libextract_dvi_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS)
 libextract_dvi_la_LDFLAGS = $(module_flags)
+libextract_dvi_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_dvi_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -374,6 +389,7 @@ libextract_raw_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS) \
 	$(GEXIV2_CFLAGS)
 libextract_raw_la_LDFLAGS = $(module_flags)
+libextract_raw_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_raw_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -391,6 +407,7 @@ libextract_gstreamer_la_CFLAGS = \
 	$(GSTREAMER_CFLAGS) \
 	$(LIBCUE_CFLAGS)
 libextract_gstreamer_la_LDFLAGS = $(module_flags)
+libextract_gstreamer_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_gstreamer_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(BUILD_LIBS) \
@@ -416,6 +433,7 @@ libextract_gif_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS) \
 	$(LIBGIF_CFLAGS)
 libextract_gif_la_LDFLAGS = $(module_flags)
+libextract_gif_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_gif_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -429,6 +447,7 @@ libextract_jpeg_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS) \
 	$(LIBJPEG_CFLAGS)
 libextract_jpeg_la_LDFLAGS = $(module_flags)
+libextract_jpeg_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_jpeg_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -442,6 +461,7 @@ libextract_tiff_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS) \
 	$(LIBTIFF_CFLAGS)
 libextract_tiff_la_LDFLAGS = $(module_flags)
+libextract_tiff_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_tiff_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(BUILD_LIBS) \
@@ -454,6 +474,7 @@ libextract_playlist_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS) \
 	$(TOTEM_PL_PARSER_CFLAGS)
 libextract_playlist_la_LDFLAGS = $(module_flags)
+libextract_playlist_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_playlist_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(BUILD_LIBS) \
@@ -464,6 +485,7 @@ libextract_playlist_la_LIBADD = \
 libextract_text_la_SOURCES = tracker-extract-text.c
 libextract_text_la_CFLAGS = $(TRACKER_EXTRACT_MODULES_CFLAGS)
 libextract_text_la_LDFLAGS = $(module_flags)
+libextract_text_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_text_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(BUILD_LIBS) \
@@ -473,6 +495,7 @@ libextract_text_la_LIBADD = \
 libextract_icon_la_SOURCES = tracker-extract-icon.c
 libextract_icon_la_CFLAGS = $(TRACKER_EXTRACT_MODULES_CFLAGS)
 libextract_icon_la_LDFLAGS = $(module_flags)
+libextract_icon_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_icon_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(BUILD_LIBS) \
@@ -485,6 +508,7 @@ libextract_iso_la_CFLAGS = \
 	$(LIBOSINFO_CFLAGS)
 
 libextract_iso_la_LDFLAGS = $(module_flags)
+libextract_iso_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_iso_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(BUILD_LIBS) \
@@ -498,6 +522,7 @@ libextract_libav_la_CFLAGS = $(TRACKER_E
 	$(AVUTIL_CFLAGS) \
 	$(AVCODEC_CFLAGS)
 libextract_libav_la_LDFLAGS = $(module_flags)
+libextract_libav_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_libav_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -513,6 +538,7 @@ libextract_bmp_la_SOURCES = tracker-extr
 libextract_bmp_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS)
 libextract_bmp_la_LDFLAGS = $(module_flags)
+libextract_bmp_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_bmp_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -524,6 +550,7 @@ libextract_dummy_la_SOURCES = tracker-ex
 libextract_dummy_la_CFLAGS = \
 	$(TRACKER_EXTRACT_MODULES_CFLAGS)
 libextract_dummy_la_LDFLAGS = $(module_flags)
+libextract_dummy_la_DEPENDENCIES = tracker-extract$(EXEEXT)
 libextract_dummy_la_LIBADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
@@ -550,6 +577,7 @@ tracker_extract_SOURCES = \
 	tracker-main.c \
 	tracker-main.h
 
+tracker_extract_LDFLAGS = -export-dynamic -Wl,--out-implib,libtracker-extract.dll.a
 tracker_extract_LDADD = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract.la \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
--- origsrc/tracker-miners-2.0.4/src/tracker-writeback/Makefile.am	2017-12-16 05:27:16.000000000 -0600
+++ src/tracker-miners-2.0.4/src/tracker-writeback/Makefile.am	2018-02-12 16:57:17.636817300 -0600
@@ -1,4 +1,4 @@
-module_flags = -module -avoid-version -no-undefined
+module_flags = -module -avoid-version -no-undefined -Wl,libtracker-writeback.exe.a
 modulesdir = $(libdir)/tracker-miners-$(TRACKER_API_VERSION)/writeback-modules
 
 AM_CPPFLAGS =                                           \
@@ -22,7 +22,8 @@ if HAVE_TAGLIB
 modules_LTLIBRARIES += libwriteback-taglib.la
 libwriteback_taglib_la_SOURCES = tracker-writeback-taglib.c
 libwriteback_taglib_la_LDFLAGS = $(module_flags)
-libwriteback_taglib_la_LIBADD = $(BUILD_LIBS) $(GLIB2_LIBS) $(TAGLIB_LIBS)
+libwriteback_taglib_la_DEPENDENCIES = tracker-writeback$(EXEEXT)
+libwriteback_taglib_la_LIBADD = $(BUILD_LIBS) $(TRACKER_WRITEBACK_LIBS) $(TAGLIB_LIBS)
 endif
 
 # XMP
@@ -30,7 +31,8 @@ if HAVE_EXEMPI
 modules_LTLIBRARIES += libwriteback-xmp.la
 libwriteback_xmp_la_SOURCES = tracker-writeback-xmp.c
 libwriteback_xmp_la_LDFLAGS = $(module_flags)
-libwriteback_xmp_la_LIBADD = $(BUILD_LIBS) $(GLIB2_LIBS) $(EXEMPI_LIBS)
+libwriteback_xmp_la_DEPENDENCIES = tracker-writeback$(EXEEXT)
+libwriteback_xmp_la_LIBADD = $(BUILD_LIBS) $(TRACKER_WRITEBACK_LIBS) $(EXEMPI_LIBS)
 endif
 
 # Playlists
@@ -55,6 +57,7 @@ tracker_writeback_SOURCES = \
 	tracker-writeback.h \
 	tracker-main.c
 
+tracker_writeback_LDFLAGS = -export-dynamic -Wl,--out-implib,libtracker-writeback.exe.a
 tracker_writeback_LDADD = \
 	$(top_builddir)/src/libtracker-miners-common/libtracker-miners-common.la \
 	$(BUILD_LIBS) \
